src/runtime/runtime.ts

```diff
@@ -1,20 +1,32 @@
+//------------------------------------------------------------------------
+// Runtime
+//------------------------------------------------------------------------
+
 type ID = number;
 type Multiplicity = number;
 
 function createHash() {
   return Object.create(null);
 }
 
+function createArray() {
```

Comment:
We use these functions during development to make it easy to track allocations. This isn't particularly important outside of the evaluation loop, but inside it limiting allocations is critical to achieving consistent (let alone good!) performance.